spec/lib/cms/audit/pages_spec.rb

```diff
@@ -0,0 +1,111 @@
+describe Cms::Audit::Pages do
+  let(:role) { :user }
+  let(:current_user) { create(:user, role: Comfy::Cms::User.roles[role]) }
+  let(:state) { 'published' }
+  let(:en_site) { create(:site) }
+  let(:cy_site) { create(:site, label: 'cy') }
+  let(:page_with_tags) { create(:page_with_multiple_tags, site: en_site, state: state) }
+  let(:page_with_meta) { create(:article_with_metadata, site: cy_site, state: state) }
+  let!(:revision) do
```

Comment:
For all this `let!` declarations (`:revision`, `revision_2`, `category` and `categorization`) as it seem to be context setup that is not reused anywhere (as far as I've seen they have no calls)... have you thought about the possibility of avoiding the boilerplate of the `let` and move them to a block where `create this in DB before executing the tests` falls as a natural thing?
```
before do
  Timecop.freeze(Time.zone.local(1990))
  create(
      :revision_with_event,
      record: page_with_tags,
      user_id: current_user.id,
      user_name: current_user.name
    )
  create(
    :revision_with_event,
    record: page_with_meta,
    user_id: current_user.id,
    user_name: current_user.name
  )
  create(
    :categorization,
    categorized_id: page_with_tags.id,
    category: create(:category, site: en_site),
    categorized_type: 'Comfy::Cms::Page'
  )
end
```
 In this way we can see explicitly what are the main thing that need to be set up in order to execute the test. While the `let` declarations outside the block are helpers/accessories to the context setup.

What do you think?

Reply:
Yep, good point! Pushing up in a minute